src/frontend/komponenter/Fagsak/Vilkårsvurdering/GeneriskVilkår/GeneriskVilkår.tsx

```diff
@@ -46,27 +42,23 @@ const GeneriskVilkår: React.FC<IProps> = ({
     visFeilmeldinger,
 }) => {
     const { erLesevisning } = useBehandling();
-    const {
-        settVilkårsvurderingFraApi,
-        settVilkårSubmit,
-        postVilkår,
-        vilkårSubmit,
-    } = useVilkårsvurdering();
+    const { settFagsak } = useFagsakRessurser();
+    const { settVilkårSubmit, postVilkår, vilkårSubmit } = useVilkårsvurdering();
 
     const [visFeilmeldingerForVilkår, settVisFeilmeldingerForVilkår] = useState(false);
     const [feilmelding, settFeilmelding] = useState('');
 
-    const håndterNyPeriodeVilkårsvurdering = (promise: Promise<Ressurs<IRestPersonResultat[]>>) => {
+    const håndterNyPeriodeVilkårsvurdering = (promise: Promise<Ressurs<IFagsak>>) => {
         promise
-            .then((nyVilkårsvurdering: Ressurs<IRestPersonResultat[]>) => {
+            .then((oppdatertFagsak: Ressurs<IFagsak>) => {
                 settVilkårSubmit(VilkårSubmit.NONE);
                 settVisFeilmeldingerForVilkår(false);
                 settFeilmelding('');
-                if (nyVilkårsvurdering.status === RessursStatus.SUKSESS) {
-                    settVilkårsvurderingFraApi(nyVilkårsvurdering.data);
-                } else if (nyVilkårsvurdering.status === RessursStatus.FEILET) {
+                if (oppdatertFagsak.status === RessursStatus.SUKSESS) {
+                    settFagsak(oppdatertFagsak);
+                } else if (oppdatertFagsak.status === RessursStatus.FEILET) {
                     settVisFeilmeldingerForVilkår(true);
-                    settFeilmelding(nyVilkårsvurdering.frontendFeilmelding);
+                    settFeilmelding(oppdatertFagsak.frontendFeilmelding);
```

Comment:
Vet ikke om det har så mye å si, men burde disse to komme i motsatt rekkefølge for å være mer sikker på at den "gamle" feilmeldingen ikke kommer til å synes?

Reply:
Ja, sikkert ikke så farlig, men kan endre.